diff --git a/dynamiccolor/color_spec.d.ts b/dynamiccolor/color_spec.d.ts
index 4ea9db132bf65d7b1d1eeb737c919a487484a785..5e98b9f95ded3363153e24a7419ba8e27cde3d34 100644
--- a/dynamiccolor/color_spec.d.ts
+++ b/dynamiccolor/color_spec.d.ts	
@@ -16,7 +16,7 @@
  */
 import { ColorSpecDelegateImpl2021 } from './color_spec_2021.js';
 import { ColorSpecDelegateImpl2025 } from './color_spec_2025.js';
-import type { DynamicColor } from './dynamic_color';
+import type { DynamicColor } from './dynamic_color.js';
 import { DynamicScheme } from './dynamic_scheme.js';
 export type SpecVersion = '2021' | '2025';
 /**
diff --git a/dynamiccolor/color_spec_2021.d.ts b/dynamiccolor/color_spec_2021.d.ts
index cc9389ba2be973878054e543d353943a10868ad2..0727f874e333caf58f666ab866a3853262e5d815 100644
--- a/dynamiccolor/color_spec_2021.d.ts
+++ b/dynamiccolor/color_spec_2021.d.ts	
@@ -16,7 +16,7 @@
  */
 import { ColorSpecDelegate } from './color_spec.js';
 import { DynamicColor } from './dynamic_color.js';
-import type { DynamicScheme } from './dynamic_scheme';
+import type { DynamicScheme } from './dynamic_scheme.js';
 /**
  * A delegate for the dynamic color spec of a DynamicScheme in the 2021 spec.
  */
diff --git a/dynamiccolor/color_spec_2025.d.ts b/dynamiccolor/color_spec_2025.d.ts
index a8b3efa8c93815782d2c2454d6d3af90aefe3c04..47661a18e9717e9824472f965d08f6889329c23c 100644
--- a/dynamiccolor/color_spec_2025.d.ts
+++ b/dynamiccolor/color_spec_2025.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { ColorSpecDelegateImpl2021 } from './color_spec_2021.js';
-import { DynamicColor } from './dynamic_color';
+import { DynamicColor } from './dynamic_color.js';
 /**
  * A delegate for the dynamic color spec of a DynamicScheme in the 2025 spec.
  */
diff --git a/dynamiccolor/color_spec_2025.js b/dynamiccolor/color_spec_2025.js
index 8bef961c7c6127c028b98ee3305270be5247a0c2..271597946422c58b91a54c1e1a748d30350ac015 100644
--- a/dynamiccolor/color_spec_2025.js
+++ b/dynamiccolor/color_spec_2025.js	
@@ -18,7 +18,7 @@ import { Hct } from '../hct/hct.js';
 import * as math from '../utils/math_utils.js';
 import { ColorSpecDelegateImpl2021 } from './color_spec_2021.js';
 import { ContrastCurve } from './contrast_curve.js';
-import { DynamicColor, extendSpecVersion } from './dynamic_color';
+import { DynamicColor, extendSpecVersion } from './dynamic_color.js';
 import { ToneDeltaPair } from './tone_delta_pair.js';
 import { Variant } from './variant.js';
 /**
diff --git a/dynamiccolor/material_dynamic_colors.d.ts b/dynamiccolor/material_dynamic_colors.d.ts
index 221cbad1d6005c8ce360868e82a2f0f9392022f8..829d5771a7a10d559da11700bb019e451df60efd 100644
--- a/dynamiccolor/material_dynamic_colors.d.ts
+++ b/dynamiccolor/material_dynamic_colors.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { DynamicColor } from './dynamic_color.js';
-import type { DynamicScheme } from './dynamic_scheme';
+import type { DynamicScheme } from './dynamic_scheme.js';
 /**
  * DynamicColors for the colors in the Material Design system.
  */
diff --git a/scheme/scheme_content.d.ts b/scheme/scheme_content.d.ts
index 1d716944a4183bbb8ad304dc5a2af5a0f8253afc..07c215bd96bd6845fa9586734826581d006b2eb6 100644
--- a/scheme/scheme_content.d.ts
+++ b/scheme/scheme_content.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /**
  * A scheme that places the source color in `Scheme.primaryContainer`.
diff --git a/scheme/scheme_content.js b/scheme/scheme_content.js
index e06c67bc68883b4a5210dc4241544ed79dec905c..29d62f7514f53d30402ee2c002801d90a2938e2f 100644
--- a/scheme/scheme_content.js
+++ b/scheme/scheme_content.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /**
  * A scheme that places the source color in `Scheme.primaryContainer`.
diff --git a/scheme/scheme_expressive.d.ts b/scheme/scheme_expressive.d.ts
index fb72fb09940ae309bd249ebe6ab6de6663cedfce..e6892443bdcf6a4c6c478d2971a3086b336e9324 100644
--- a/scheme/scheme_expressive.d.ts
+++ b/scheme/scheme_expressive.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /**
  * A Dynamic Color theme that is intentionally detached from the source color.
diff --git a/scheme/scheme_expressive.js b/scheme/scheme_expressive.js
index 43d05c6a9989566f2f300e8bf01fa4f1a8e120f0..baf7ca348f18ddc82e8d1f4df26161b70fbb578d 100644
--- a/scheme/scheme_expressive.js
+++ b/scheme/scheme_expressive.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /**
  * A Dynamic Color theme that is intentionally detached from the source color.
diff --git a/scheme/scheme_fidelity.d.ts b/scheme/scheme_fidelity.d.ts
index ce8cd7fccc00ad02bcc71e2ced989710f4f56a15..967636164eea86399bd667151b4671ee1e0c5c0d 100644
--- a/scheme/scheme_fidelity.d.ts
+++ b/scheme/scheme_fidelity.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /**
  * A scheme that places the source color in `Scheme.primaryContainer`.
diff --git a/scheme/scheme_fidelity.js b/scheme/scheme_fidelity.js
index a7461cdd4b49a49e642cd1060fba95ea1b6c7a1e..602d1eec140103f4f9501d9a84238eece61416f9 100644
--- a/scheme/scheme_fidelity.js
+++ b/scheme/scheme_fidelity.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /**
  * A scheme that places the source color in `Scheme.primaryContainer`.
diff --git a/scheme/scheme_fruit_salad.d.ts b/scheme/scheme_fruit_salad.d.ts
index df109ccc78e9182193c2ba7a2a0a215a10c8a0ba..d3fad5e033d60cb49773ca4448f4ec4e2440c89a 100644
--- a/scheme/scheme_fruit_salad.d.ts
+++ b/scheme/scheme_fruit_salad.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /**
  * A playful theme - the source color's hue does not appear in the theme.
diff --git a/scheme/scheme_fruit_salad.js b/scheme/scheme_fruit_salad.js
index 87443afa8bb09d9c6f67be12fde55b7db9b9f37a..aeaff555801de637d3b8005eed30718d88c362df 100644
--- a/scheme/scheme_fruit_salad.js
+++ b/scheme/scheme_fruit_salad.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /**
  * A playful theme - the source color's hue does not appear in the theme.
diff --git a/scheme/scheme_monochrome.d.ts b/scheme/scheme_monochrome.d.ts
index a52e962ac77a6a1cc6397864d3119e94ea2d110c..bd9f3a0aa5d7233cbce92f2430765622ef5732c1 100644
--- a/scheme/scheme_monochrome.d.ts
+++ b/scheme/scheme_monochrome.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /** A Dynamic Color theme that is grayscale. */
 export declare class SchemeMonochrome extends DynamicScheme {
diff --git a/scheme/scheme_monochrome.js b/scheme/scheme_monochrome.js
index 30ad712ad104e397788ff3d8e11ea28ca00c0533..d18d656c4f98bbd4a1340feeaeff390f9a8b83fc 100644
--- a/scheme/scheme_monochrome.js
+++ b/scheme/scheme_monochrome.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /** A Dynamic Color theme that is grayscale. */
 export class SchemeMonochrome extends DynamicScheme {
diff --git a/scheme/scheme_neutral.d.ts b/scheme/scheme_neutral.d.ts
index 733d46f8bc469555242e531c00eb459317999c55..a2146ba33080946cc9a6c63610ab354e10905044 100644
--- a/scheme/scheme_neutral.d.ts
+++ b/scheme/scheme_neutral.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /** A Dynamic Color theme that is near grayscale. */
 export declare class SchemeNeutral extends DynamicScheme {
diff --git a/scheme/scheme_neutral.js b/scheme/scheme_neutral.js
index 0d03a5e0f0200feb471860b48d5243dd5da5429d..e4cffaaa9d66bd056fded64bfb0ce43341789c66 100644
--- a/scheme/scheme_neutral.js
+++ b/scheme/scheme_neutral.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /** A Dynamic Color theme that is near grayscale. */
 export class SchemeNeutral extends DynamicScheme {
diff --git a/scheme/scheme_rainbow.d.ts b/scheme/scheme_rainbow.d.ts
index 582f8dc336610849201b06c9fe998256973d56d5..91194997645431fd921e44c4328dda3b27d791be 100644
--- a/scheme/scheme_rainbow.d.ts
+++ b/scheme/scheme_rainbow.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /**
  * A playful theme - the source color's hue does not appear in the theme.
diff --git a/scheme/scheme_rainbow.js b/scheme/scheme_rainbow.js
index 65e6d3fd934ed07a2e6efdf6c552c306873d26e2..f80be9ae6da0d181c18a064ca4c645835abf9f86 100644
--- a/scheme/scheme_rainbow.js
+++ b/scheme/scheme_rainbow.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /**
  * A playful theme - the source color's hue does not appear in the theme.
diff --git a/scheme/scheme_tonal_spot.d.ts b/scheme/scheme_tonal_spot.d.ts
index 65033b308a63d0486ef4e44d361ffccc6c1e734e..d06fe6cc3b87378bca1fb8d5f1049cc0fcfefb2a 100644
--- a/scheme/scheme_tonal_spot.d.ts
+++ b/scheme/scheme_tonal_spot.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /**
  * A Dynamic Color theme with low to medium colorfulness and a Tertiary
diff --git a/scheme/scheme_tonal_spot.js b/scheme/scheme_tonal_spot.js
index 6c506c3c23279e5842757b991cd066ef266fceea..c48f498c41e0cde98f34c3d7394a75ef1bf764cd 100644
--- a/scheme/scheme_tonal_spot.js
+++ b/scheme/scheme_tonal_spot.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /**
  * A Dynamic Color theme with low to medium colorfulness and a Tertiary
diff --git a/scheme/scheme_vibrant.d.ts b/scheme/scheme_vibrant.d.ts
index 676d54f7d3d62e91e3d303ab2fd6f8fd38d3abda..00e25cb010c42c39019da615c92343ea3c8e94c0 100644
--- a/scheme/scheme_vibrant.d.ts
+++ b/scheme/scheme_vibrant.d.ts	
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 import { SpecVersion } from '../dynamiccolor/color_spec.js';
-import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme, Platform } from '../dynamiccolor/dynamic_scheme.js';
 import { Hct } from '../hct/hct.js';
 /**
  * A Dynamic Color theme that maxes out colorfulness at each position in the
diff --git a/scheme/scheme_vibrant.js b/scheme/scheme_vibrant.js
index cba1172e7d7ccae62d03e635c38063153bc5a61a..3a37230ad4b589c7d3071dbb093335eab0d71f4c 100644
--- a/scheme/scheme_vibrant.js
+++ b/scheme/scheme_vibrant.js	
@@ -14,7 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { DynamicScheme } from '../dynamiccolor/dynamic_scheme';
+import { DynamicScheme } from '../dynamiccolor/dynamic_scheme.js';
 import { Variant } from '../dynamiccolor/variant.js';
 /**
  * A Dynamic Color theme that maxes out colorfulness at each position in the
